### ISSUE TITLE
Download stuck at 0KB/s

### ISSUE THREAD

**Issue by numlockkey** [FaisalUmair/udemy-downloader-gui#315]
Latest 1.8 version. Auto retry on error is checked.

![2019-10-12](https://user-images.githubusercontent.com/56490338/66710522-5fab4980-ed48-11e9-83c7-3677389f9b09.png)

**Comment by FaisalUmair** [comment]
How long did you wait for it? If it is in auto retry mode, it will keep retrying.   And I see this in 'Downloads' section, is the status there by default when you start it or is it stuck?

**Comment by numlockkey** [comment]
I waited about 15-20 minutes. The first few files downloaded OK and then it got stuck.

**Comment by akhairy877** [comment]
![0KB](https://user-images.githubusercontent.com/46758725/66773882-65c43600-eec0-11e9-9301-4838fd26066c.jpg)

**Comment by FaisalUmair** [comment]
@numlockkey  Does it happen with a particular course only? What happens if you turn off the auto retry option and delete the course files?

@akhairy877 Do you have auto retry option enabled?

**Comment by akhairy877** [comment]
Yes, I have auto retry option enabled and the latest 1.8 version

**Comment by FaisalUmair** [comment]
> @FaisalUmair Yes, I have auto retry option enabled and the latest 1.8 version!

Can you try disabling that and let me know what happens?

**Comment by akhairy877** [comment]
@FaisalUmair It's failing quickly after disabling this option!

**Comment by FaisalUmair** [comment]
@akhairy877  Just try deleting the course file and retry download. See if it fails again?

**Comment by akhairy877** [comment]
@FaisalUmair I made this many times before and without any change.

**Comment by FaisalUmair** [comment]
So it's failing again?

Can you try using an old version of Udeler. And see if that is able to download this course. Actually the auto retry feature will keep on retrying (it was added to help retrying downloads with expired links). Looks like Udeler is unable to start the download, it is failing and auto retry is trying again and again, making it look it is stuck.

I would like to see if the download is failing because of any other reason. So you can try to use old version of Udeler  (preferably 1.7.0) and download the same course.

**Comment by akhairy877** [comment]
@FaisalUmair I tried version 1.7.0 to download the same course now after deleting the course folder and it's still failing!

**Comment by numlockkey** [comment]
OK, I deleted the partially downloaded course and then restarted the download -- it's seems to be doing OK right now. Still, this is a 100+GB course and having to delete GBs of data just to start all over again is not a good solution.

**Comment by akhairy877** [comment]
@FaisalUmair Could you please tell me how can I download this course now because I need it urgently?

**Comment by FaisalUmair** [comment]
@akhairy877 It might be something specific to this course only, are you able to download other courses?

**Comment by akhairy877** [comment]
@FaisalUmair I tried to download 2 courses before trying to download this course and they failed from their beginning also!

**Comment by MdMozammilAnwar** [comment]
@akhairy877 , @FaisalUmair  i m facing the exact problem , which @akhairy877  is facing ,
i have tried in 1.6 version, 1.7 version, 1.8 version , it is not working .
earlier it was working in 1.7,some time i was getting the download failure problem.
but now since 2 days i am facing the download speed became too much slow , and just after 1% or 2% download is failing .
@FaisalUmair please bring the solution , u r one and only hope for us.

**Comment by akhairy877** [comment]
@MdMozammilAnwar @FaisalUmair Finally, I found the solution which is adding Udeler to my firewall and antivirus trusted applications and exclusions lists. I wish it will solve your problem @MdMozammilAnwar

**Comment by FaisalUmair** [comment]
@akhairy877 Great you found a solution. Firewalls can block applications from using network. But the original issue of this thread is not related to firewall I guess. You were facing same problem in earlier versions, so for you it was to do with firewall.

@MdMozammilAnwar You can also try to disable any firewall and antivirus protection and see if it works for you.

**Comment by MdMozammilAnwar** [comment]
@FaisalUmair  Allah bless you bro.
Millions of students are  getting benefited  to Udeler .
You have developed an amazing  product .

**Comment by FaisalUmair** [comment]
@MdMozammilAnwar Thank you so much for your kind words of appreciation.

**Comment by mmkholy7** [comment]
Alsalam Alikom @FaisalUmair  Great Application, I wish I can build cool stuff like you do ! Kudos sir! 
I am facing a similar issue, I was able to download some courses and some others they are not downloading. Any idea what the problem could be?
